zerver/lib/typed_endpoint.py

```diff
@@ -171,26 +207,45 @@ def parse_single_parameter(
     # otherwise causes undesired behaviors that the annotated metadata gets
     # lost. This is fixed in Python 3.11:
     # https://github.com/python/cpython/issues/90353
-    if param_default is None:
-        origin = get_origin(param_type)
+    if (
+        sys.version_info < (3, 11) and param_default is None
+    ):  # nocoverage # We lose coverage of this with Python 3.11+ only
```

Comment:
Hmm, is the `version_info` check correct? I think we want someone developing on Python 3.11 to not use nesting constructions in their code, since it'll break on other major Python versions...